docker環境の構築

halki117 opened the pull request:
dockerで環境構築しました。作成ファイルは以下です。
・Dockerfile
・docker-compose.yml
・.dockerignore

halki117 committed:
docker環境の構築

k-iwahara commented:
以下の点の修正をお願いします！
- 案件はnpmではなくてyarn使っているので、yarnにしましょう！
- mysqlのバージョンは8系なので、そこも合わせて8系にしましょう！
- DBの接続情報などは環境変数のファイルで管理しましょう！

halki117 committed:
Dockerfile等修正

・npmからyarnへ変更
・mysqlのバージョンを8系へ変更
・DBの接続情報を.envで管理（.envはgitの管理外としている）

halki117 committed:
hbs（テンプレートエンジン）導入

halki117 committed:
ライブラリー導入　typeorm @nestjs/typeorm mysql2

yarn add typeorm @nestjs/typeorm mysql2

halki117 committed:
ts-node更新

halki117 committed:
ユーザーEntityの作成、migrationファイル(usersテーブル)作成及び実行

halki117 committed:
認証系ライブラリーの追加

yarn add @nestjs/passport @nestjs/jwt passport passport-jwt bcrypt

halki117 committed:
認証サービス(auth.service , user.service)の作成

halki117 committed:
認証用コントローラー、gurdの作成

halki117 committed:
Passportストラテジーの設定など

halki117 committed:
ログアウト処理の実装

halki117 committed:
module記述修正

以下エラーに対する対応
ERROR [ExceptionHandler] Nest can't resolve dependencies of the UserService (?). Please make sure that the argument UserRepository at index [0] is available in the AppModule context.

AuthServiceとUserServiceをAppModuleのprovidersから削除する

halki117 committed:
新規登録画面、ログイン画面view（叩き台）作成

halki117 committed:
投稿機能関係のcontroller, service, module作成

halki117 committed:
投稿作成・みんなの投稿画面（叩き台）作成

halki117 committed:
viewレイアウト調整

halki117 committed:
ヘッダー、各viewファイルへのpathの設定

halki117 committed:
フロント側、フォームタグのpath記載

halki117 committed:
lacalストラテジー編集（emailとpasswordでログインする仕様に変更）

halki117 committed:
Cookieへjwt保存、CookieからJWTの取得、JwtAuthGuardへアクセス保護のロジック記載

参考：https://qiita.com/biga816/items/bc0fb17725f9dc2b8525

ライブラリー
"cookie-parser": "^1.4.6",

halki117 committed:
Update data-source.ts

halki117 committed:
新規登録、及びログイン後に投稿一覧へ遷移

halki117 committed:
未ログイン状態の際にログイン画面へのリダイレクト（JwtAuthガード修正）

halki117 committed:
ログアウト後にログイ画面へ遷移

halki117 committed:
テンプレートエンジンejsに変更、hederとfooter部を共通テンプレートとして切り分け

halki117 committed:
ログイン済み、未ログインでのヘッダー表示の切り替え

halki117 committed:
ドロップダウンメニュー実装

halki117 committed:
Merge pull request #2 from halki117/node_subject4-1

Node課題4-1

halki117 committed:
投稿作成・編集・一覧の静的レイアウト作成

halki117 committed:
nest/cli追加

halki117 committed:
Postエンティティー•migrationファイル作成、Userモデルとのリレーションの実装

halki117 committed:
Postモジュールへのimports追記

halki117 committed:
クラスバリデーション実装（ class-validatorインストール post.dto作成）

halki117 committed:
DTOを使ったバリデーションを有効化

halki117 committed:
投稿作成機能(create)実装

halki117 committed:
投稿作成機能(create)実装 修正

halki117 committed:
css修正

halki117 committed:
view修正

halki117 committed:
投稿編集画面へのデータ繋ぎ込み

halki117 committed:
投稿更新機能の実装

halki117 committed:
投稿削除機能の実装

halki117 committed:
不要コメントアウト削除

halki117 committed:
Restに沿ってURIの命名を変更

halki117 committed:
URIの変更によるリダイレクト先の変更

halki117 committed:
uri, リダイレクトpathの修正

halki117 committed:
Merge pull request #3 from halki117/node_subject4-2

Node課題4-2
